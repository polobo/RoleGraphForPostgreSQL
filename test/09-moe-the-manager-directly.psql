-- ADMIN OPTION isn't transitive by itself
-- Moe the Service Manager directly administers manager, which directly
-- adminsters advisors, which directly administers techs, but moe,
-- lacking set/inherit to the other, is not recorded as an administrator
-- to the others.
\if :is_su
    SELECT format('CREATE USER u9_moe_%s NOINHERIT;', :'ctx') \gexec
    SELECT format('CREATE GROUP g9a_service_manager_%s NOINHERIT;', :'ctx') \gexec
    SELECT format('CREATE GROUP g9b_service_advisor_%s NOINHERIT;', :'ctx') \gexec
    SELECT format('CREATE GROUP g9c_service_tech_%s NOINHERIT;', :'ctx') \gexec
    SELECT format('GRANT g9a_service_manager_%1$s TO u9_moe_%1$s WITH SET FALSE, ADMIN OPTION;', :'ctx') \gexec
    SELECT format('GRANT g9b_service_advisor_%1$s TO g9a_service_manager_%1$s WITH SET FALSE, ADMIN OPTION;', :'ctx') \gexec
    SELECT format('GRANT g9c_service_tech_%1$s TO g9b_service_advisor_%1$s WITH SET FALSE, ADMIN OPTION;', :'ctx') \gexec
\endif
