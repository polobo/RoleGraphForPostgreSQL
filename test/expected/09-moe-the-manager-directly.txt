                                             Test Result for 09-moe-the-manager-directly
        rolname         |              administration              | memberof_users | memberof_groups | member_users | member_groups 
------------------------+------------------------------------------+----------------+-----------------+--------------+---------------
 g9a_service_manager_su | of g9b_service_advisor_su from superuser+|                |                 |              | 
                        | by u9_moe_su from superuser              |                |                 |              | 
 g9b_service_advisor_su | of g9c_service_tech_su from superuser   +|                |                 |              | 
                        | by g9a_service_manager_su from superuser |                |                 |              | 
 g9c_service_tech_su    | by g9b_service_advisor_su from superuser |                |                 |              | 
 cr_admin               | Create Role                              |                |                 |              | 
 u9_moe_su              | of g9a_service_manager_su from superuser |                |                 |              | 
(5 rows)

